package/harvester-os/files/usr/sbin/harv-install

```diff
@@ -185,34 +215,64 @@ preload_rke2_images()
       export PATH=/var/lib/rancher/rke2/bin:$PATH
       export CONTAINERD_ADDRESS=/run/k3s/containerd/containerd.sock
 
-      # wait for containerd to be ready
-      until ctr --connect-timeout 1s version&>/dev/null
-      do
-        sleep 1
-      done
+      wait_for_containerd_ready
+
+      if [ "$install_mode" = "ISO" ]; then
+        echo "Stop rke2 and remove temporary rke2 files..."
+        pkill rke2
+        rm /rke2.log
+        rm /etc/rancher/rke2/rke2.yaml
+        rm -rf /var/lib/rancher/rke2/server
+        rm -rf /var/lib/rancher/rke2/agent/pod-manifests/*
+
+        echo "Start containerd..."
+        containerd -c /var/lib/rancher/rke2/agent/etc/containerd/config.toml -a $CONTAINERD_ADDRESS --state /run/k3s/containerd --root /var/lib/rancher/rke2/agent/containerd &> /containerd.log &
+
+        wait_for_containerd_ready
+
+        # load images
+        for i in $tmp_images_dir/*.tar.zst; do
+            echo "Load images from $i"
+            zstd -d $i -o /usr/local/images.tar
+            ctr -n k8s.io images import --no-unpack /usr/local/images.tar
+            rm /usr/local/images.tar
+        done
+      fi
 
       # make sure all preloading images are ready
-      for i in /tmp/images-lists/*.txt; do
+      for i in $images_lists_dir/*.txt; do
         stdbuf -oL ctr-check-images.sh $i
       done
 
-      # stop containerd
-      pkill rke2
-      rm /rke2.log
-      rm /etc/rancher/rke2/rke2.yaml
-
-      rm -rf /var/lib/rancher/rke2/server
-      rm -rf /var/lib/rancher/rke2/agent/pod-manifests/*
+      # tearing down containerd/rke2
+      if [ "$install_mode" = "ISO" ]; then
+        echo "Stop containerd..."
+        pkill containerd
+        rm /containerd.log
+      else
+        echo "Stop rke2..."
```

Comment:
```suggestion
        echo "Stop RKE2..."
```

Just a small nit to upper case the word in the script.

Reply:
Updated, thanks!